diff --git a/README.md b/README.md
index c9b9a3d..eb6d14c 100644
--- a/README.md
+++ b/README.md
@@ -92,4 +92,13 @@ https://www.youtube.com/watch?v=vs75_PiET6g
 Link to the screencast is:
 ```
 https://www.youtube.com/watch?v=19e_F-gD4X0
+```
+
+#P4 
+Command to use in VM: cat /proc/fairscheduler
+
+##Screencast
+Link to the screencast is:
+```
+https://www.youtube.com/watch?v=c7hDNmfgxlA
 ```
\ No newline at end of file
diff --git a/linux-2.6.22.19-cs543/fs/proc/proc_misc.c b/linux-2.6.22.19-cs543/fs/proc/proc_misc.c
index b1ecfdf..fa64b3f 100644
--- a/linux-2.6.22.19-cs543/fs/proc/proc_misc.c
+++ b/linux-2.6.22.19-cs543/fs/proc/proc_misc.c
@@ -52,6 +52,7 @@
 #include <asm/tlb.h>
 #include <asm/div64.h>
 #include "internal.h"
+#include <linux/sched.h>
 
 #define LOAD_INT(x) ((x) >> FSHIFT)
 #define LOAD_FRAC(x) LOAD_INT(((x) & (FIXED_1-1)) * 100)
@@ -631,6 +632,19 @@ static int show_stat(struct seq_file *p, void *v)
 
 	return 0;
 }
+int fair_scheduler_information(char *buf,char **start,off_t offset,int count,int *eof,void *data )
+{
+	int len=0;
+	struct task_struct *procs;
+
+
+	for_each_process(procs) {
+
+       		len  += sprintf(buf+len, "\nUserID: %d \nProcess ID: %d \nTimeslice: %d \n",procs->user->uid, procs->pid, procs->time_slice);
+  	}
+  
+	return len;
+}
 
 static int stat_open(struct inode *inode, struct file *file)
 {
@@ -789,6 +803,7 @@ void __init proc_misc_init(void)
 		{"cmdline",	cmdline_read_proc},
 		{"locks",	locks_read_proc},
 		{"execdomains",	execdomains_read_proc},
+		{"fairscheduler", fair_scheduler_information},
 		{NULL,}
 	};
 	for (p = simple_ones; p->name; p++)
